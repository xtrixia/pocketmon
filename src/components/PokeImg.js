import { css } from "@emotion/react";
import clsx from "clsx";

import { BREAKPOINTS } from "../root/breakpoints";

const xsmall = css`
  width: 25px;
  @media (min-width: ${BREAKPOINTS.sm}) {
    width: 50px;
  }
`;

const small = css`
  width: 100px;
  @media (min-width: ${BREAKPOINTS.sm}) {
    width: 250px;
  }
`;

const medium = css`
  width: 250px;
  @media (min-width: ${BREAKPOINTS.sm}) {
    width: 300px;
  }
  @media (min-width: ${BREAKPOINTS.lg}) {
    width: 350px;
  }
`;

function PokeImg({ alt, background, className, img, type, ...props }) {
  let imgType = "";

  switch (type) {
    case "small":
      imgType = small;
      break;
    case "medium":
      imgType = medium;
      break;
    default:
      imgType = xsmall;
      break;
  }
  return (
    <img
      src={img}
      alt={alt}
      className={clsx(
        [imgType],
        className,
        css`
          background: ${background};
        `
      )}
      {...props}
    />
  );
}

export default PokeImg;
