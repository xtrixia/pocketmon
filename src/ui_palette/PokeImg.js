import { css } from "@emotion/css";
import clsx from "clsx";

import { BREAKPOINTS } from "../root/breakpoints";

const xsmall = css`
  width: 50px;
`;

const small = css`
  width: 100px;
  @media (min-width: ${BREAKPOINTS.sm}) {
    width: 250px;
  }
`;

const medium = css`
  width: 250px;
  @media (min-width: ${BREAKPOINTS.sm}) {
    width: 300px;
  }
  @media (min-width: ${BREAKPOINTS.lg}) {
    width: 350px;
  }
`;

const large = css`
  width: 350px;
`;

function PokeImg({ alt, background, className, img, type, ...props }) {
  let imgType = large;

  switch (type) {
    case "small":
      imgType = small;
      break;
    case "medium":
      imgType = medium;
      break;
    default:
      imgType = xsmall;
      break;
  }
  return (
    <img
      src={img}
      alt={alt}
      className={clsx(
        [imgType],
        className,
        css`
          background: ${background};
        `
      )}
      {...props}
    />
  );
}

export default PokeImg;
