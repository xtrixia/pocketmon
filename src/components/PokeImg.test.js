import React from "react";
import renderer from "react-test-renderer";

import PokeImg from "./PokeImg";

test("PokeImg render correctly", () => {
  const props = {
    src: "https://cdn.logo.com/hotlink-ok/logo-social.png",
    alt: "something",
  };
  const component = renderer.create(<PokeImg {...props} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.type).toStrictEqual("img");
  expect(tree.props.src).toStrictEqual(props.src);
  expect(tree.props.alt).toStrictEqual(props.alt);
});
