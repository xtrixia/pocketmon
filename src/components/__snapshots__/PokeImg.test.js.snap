// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeImg render correctly 1`] = `
<img
  alt="something"
  className="css-1vwu78b css-owkuzg"
  src="https://cdn.logo.com/hotlink-ok/logo-social.png"
/>
`;
